### PR TITLE
[BACKLOG-5353] Remove deprecated ThinDriver and related classes from kettle core

### PR DESCRIPTION
**Pull request by PavelSakoun**
Needs to be merged along/after https://github.com/pentaho/pentaho-platform-ee/pull/921

### CODE REVIEW

**Comment by lucboudreau** [comment]
http://10.100.11.222:8080/job/wingman-osgi/74/

**Comment by lucboudreau** [comment]
Error analyzing output with org.pentaho.build.buddy.bundles.analyzer.header.LicenseHeaderAnalyzer@57b5a6e0: /home/devuser/docker-build/assembly/build/assembly-0.0.1/source/pentaho-kettle-head-2346-3ee27bfb-9f3b-4062-a281-d0103f3d83d2/core/src/org/pentaho/di/core/jdbc/FieldVariableMapping.java (No such file or directory)

File: core/src/org/pentaho/di/core/sql/SQLUtil.java [hunk 1]
```diff
@@ -48,14 +38,17 @@
...
 
-/**
- * Static methods provided by this class should be copied to their respective projects
- *
- * Data Service client code is now available in the pdi-dataservice-plugin project
- *
- */
-@Deprecated
-public class ThinUtil {
+import java.math.BigDecimal;
```
Comment (mkambol) [inline core/src/org/pentaho/di/core/sql/SQLUtil.java]: @PavelSakoun Don't dataservices server and client use ThinUtil?  
Should this class be moved to dataservices client?

Comment (PavelSakoun) [inline core/src/org/pentaho/di/core/sql/SQLUtil.java]: Missed it's usage in dataservices projects. It's actually used in kettle-core mostly and I'd keep it there, don't like the idea of making kettle-core depend on dataservices.

Comment (mkambol) [inline core/src/org/pentaho/di/core/sql/SQLUtil.java]: What's the use in kettle-core?  Is it just for other stuff that's only used by dataservices?

**Comment by brosander** [comment]
Build Commands:
==============
Build:
------
1. ant -Dtestreports.dir=bin/reports/unit-test -Djunit.jvmargs=-XX:PermSize=256m -f core/build.xml clean-all resolve jacoco && ant -Dpentaho.coding.standards.version=1.0.4 -Dtestreports.dir=bin/reports/integration-test -Djunit.jvmargs=-XX:PermSize=256m -f core/build.xml jacoco-integration checkstyle publish-local
2. ant -Dtestreports.dir=bin/reports/unit-test -Djunit.jvmargs=-XX:PermSize=256m -f engine/build.xml clean-all resolve jacoco && ant -Dpentaho.coding.standards.version=1.0.4 -Dtestreports.dir=bin/reports/integration-test -Djunit.jvmargs=-XX:PermSize=256m -f engine/build.xml jacoco-integration checkstyle publish-local

Cleanup:
-------
rm -r ~/.ivy2/local || echo no publish local to remove



Changed files:
==============
  core/src/org/pentaho/di/core/jdbc/FieldVariableMapping.java
  core/src/org/pentaho/di/core/jdbc/RowsResultSet.java
  core/src/org/pentaho/di/core/jdbc/ThinConnection.java
  core/src/org/pentaho/di/core/jdbc/ThinDatabaseMetaData.java
  core/src/org/pentaho/di/core/jdbc/ThinDriver.java
  core/src/org/pentaho/di/core/jdbc/ThinParameterMetaData.java
  core/src/org/pentaho/di/core/jdbc/ThinPreparedStatement.java
  core/src/org/pentaho/di/core/jdbc/ThinResultSet.java
  core/src/org/pentaho/di/core/jdbc/ThinResultSetMetaData.java
  core/src/org/pentaho/di/core/jdbc/ThinServiceInformation.java
  core/src/org/pentaho/di/core/jdbc/ThinStatement.java
  core/src/org/pentaho/di/core/jdbc/TransDataService.java
  core/src/org/pentaho/di/core/sql/FoundClause.java
  core/src/org/pentaho/di/core/sql/IifFunction.java
  core/src/org/pentaho/di/core/sql/SQL.java
  core/src/org/pentaho/di/core/sql/SQLCondition.java
  core/src/org/pentaho/di/core/sql/SQLField.java
  core/src/org/pentaho/di/core/sql/SQLFields.java
  core/src/org/pentaho/di/core/sql/SQLUtil.java
  core/test-src/org/pentaho/di/core/jdbc/SQLTest.java
  core/test-src/org/pentaho/di/core/jdbc/ThinDatabaseMetaDataTest.java
  core/test-src/org/pentaho/di/core/jdbc/ThinPreparedStatementTest.java
  core/test-src/org/pentaho/di/core/sql/SQLTest.java
  core/test-src/org/pentaho/di/core/sql/SQLUtilTest.java
  engine/src/org/pentaho/di/www/SlaveServerConfig.java



Unit test coverage change:
============
org.pentaho.di.core.jdbc.FoundClause:
 * Complexity Change : **-100.00%**
 * Instruction Change : **-100.00%**
 * Line Change : **-100.00%**
 * Method Change : **-100.00%**

org.pentaho.di.core.jdbc.RowsResultSet:
 * Branch Change : **-20.00%**
 * Complexity Change : **-3.86%**
 * Instruction Change : **-7.50%**
 * Line Change : **-6.54%**
 * Method Change : **-2.60%**

org.pentaho.di.core.jdbc.ThinDatabaseMetaData:
 * Branch Change : **-23.33%**
 * Complexity Change : **-6.63%**
 * Instruction Change : **-53.73%**
 * Line Change : **-37.66%**
 * Method Change : **-6.63%**

org.pentaho.di.core.jdbc.ThinPreparedStatement:
 * Branch Change : **-15.56%**
 * Complexity Change : **-18.39%**
 * Instruction Change : **-29.87%**
 * Line Change : **-26.19%**
 * Method Change : **-20.97%**

org.pentaho.di.core.jdbc.ThinResultSetMetaData:
 * Branch Change : **-11.76%**
 * Complexity Change : **-12.82%**
 * Instruction Change : **-17.65%**
 * Line Change : **-20.45%**
 * Method Change : **-16.67%**

org.pentaho.di.core.jdbc.ThinStatement:
 * Complexity Change : **-3.64%**
 * Instruction Change : **-3.12%**
 * Line Change : **-5.97%**
 * Method Change : **-4.26%**

org.pentaho.di.core.jdbc.ThinUtil:
 * Branch Change : **-82.95%**
 * Complexity Change : **-73.65%**
 * Instruction Change : **-90.95%**
 * Line Change : **-89.04%**
 * Method Change : **-93.10%**

org.pentaho.di.core.sql.FoundClause:
 * Complexity New Coverage : 100.00%
 * Instruction New Coverage : 100.00%
 * Line New Coverage : 100.00%
 * Method New Coverage : 100.00%

org.pentaho.di.core.sql.SQLUtil:
 * Branch New Coverage : 82.95%
 * Complexity New Coverage : 73.65%
 * Instruction New Coverage : 90.95%
 * Line New Coverage : 89.04%
 * Method New Coverage : 93.10%

org.pentaho.di.core.auth.core.impl.DefaultAuthenticationConsumerFactory:
 * Branch Change :  + 7.14%
 * Complexity Change :  + 7.69%

org.pentaho.di.job.entries.checkdbconnection.JobEntryCheckDbConnections:
 * Branch Change :  + 1.23%
 * Complexity Change :  + 1.67%
 * Instruction Change :  + .32%
 * Line Change :  + 1.11%

org.pentaho.di.job.entries.sql.JobEntrySQL:
 * Instruction Change : **-1.64%**
 * Line Change : **-1.94%**

org.pentaho.di.repository.kdr.KettleDatabaseRepositoryCreationHelper:
 * Branch Change : **-1.64%**
 * Complexity Change : **-.69%**
 * Instruction Change : **-2.43%**
 * Line Change : **-3.70%**

org.pentaho.di.trans.steps.excelwriter.ExcelWriterStepMeta:
 * Instruction Change :  + .08%

org.pentaho.di.trans.steps.replacestring.ReplaceStringMeta:
 * Instruction Change :  + .07%

org.pentaho.di.trans.steps.sortedmerge.SortedMergeMeta:
 * Instruction Change :  + .18%

org.pentaho.di.trans.steps.symmetriccrypto.symmetriccryptotrans.SymmetricCryptoTransMeta:
 * Branch Change :  + 3.33%
 * Complexity Change :  + 1.85%
 * Instruction Change :  + .28%

org.pentaho.di.trans.steps.uniquerows.UniqueRowsMeta:
 * Instruction Change : **-.18%**

org.pentaho.di.www.SlaveServerConfig:
 * Branch Change :  + 1.56%
 * Complexity Change :  + .66%
 * Instruction Change :  + .38%
 * Line Change :  + .42%
 * Method Change :  + .26%

**Pull request closed without merging**